src/mjolnir/landmark_database_builder.cc

```diff
@@ -0,0 +1,189 @@
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <iostream>
+
+#include "baldr/graphconstants.h"
+#include "filesystem.h"
+#include "mjolnir/landmark_database_builder.h"
+#include "mjolnir/util.h"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+bool LandmarkDatabase::openDatabase() {
```

Comment:
this body of this function and the function below should both be put in the constructor for the class. it doesnt make a lot of sense to have the person who instantiates the db tell it to do this stuff that has to be done regardless, lets just do it automatically.

now it may be the case that we have to differentiate between opening for the first time and opening for reading/appending but i kind of think there is very little if any work to make that happen

Reply:
indexing should also happen automatically